excel/src/main/java/org/apache/metamodel/excel/DefaultSpreadsheetReaderDelegate.java

```diff
@@ -167,50 +166,114 @@ private MutableTable createTable(final Workbook wb, final Sheet sheet) {
                     break;
                 }
             }
-
             if (hasColumns) {
-                createColumns(table, wb, row);
+                createColumns(table, wb, row, columnTypes);
             }
         }
-
         return table;
     }
 
+    private void setColumnType(Iterator<Row> data, int rowLength, ColumnType[] columnTypes) {
+        while (data.hasNext()) {
+            Row row = data.next();
+            for (int index = 0; index < rowLength; index++) {
+                if (row.getLastCellNum() == 0) {
+                    continue;
+                }
+                if (row.getCell(index) == null) {
+                    columnTypes = checkColumnType(ColumnType.STRING, columnTypes, index);
+                } else {
+                    CellType cellType = row.getCell(index).getCellType();
+                    if (cellType.getCode() != 0 && cellType.getCode() <= 2) {
+                        columnTypes = checkColumnType(ColumnType.STRING, columnTypes, index);
+                    } else if (cellType.getCode() == 0) {
+                        columnTypes = checkColumnType((row.getCell(index).getNumericCellValue() % 1 == 0)
```

Comment:
In case of a Numeric cell type, can you use some of poi's util classes to more specifically determine its type? E.g. `DateUtil#isCellDateFormatted(Cell)` to see if it's a `Date`?

Reply:
done with next commit